### PR TITLE
Search by mobile, phonenumber and alt_name

### PR DESCRIPTION
**Pull request by SebaMutuku**

### CODE REVIEW

File: src/main/java/org/opensrp/web/rest/SearchResource.java [hunk 1]
```diff
@@ -1,41 +1,37 @@
...
+import javax.servlet.http.HttpServletRequest;
+import java.text.ParseException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static org.opensrp.common.AllConstants.BaseEntity.LAST_UPDATE;
+import static org.opensrp.common.AllConstants.Client.*;
```
Comment (dubdabasoduba) [inline src/main/java/org/opensrp/web/rest/SearchResource.java:32]: Please avoid importing `*`

File: src/main/java/org/opensrp/web/rest/RestUtils.java [hunk 2]
```diff
@@ -27,10 +11,20 @@
 import org.opensrp.domain.Multimedia;
 import org.opensrp.service.multimedia.MultimediaFileManager;
 import org.opensrp.service.multimedia.S3MultimediaFileManager;
-import org.slf4j.LoggerFactory;
 import org.springframework.http.HttpHeaders;
 import org.springframework.security.core.Authentication;
 
+import javax.servlet.http.HttpServletRequest;
+import java.io.*;
```
[the remark this answers is not in the document]
Comment (SebaMutuku) [reply src/main/java/org/opensrp/web/rest/RestUtils.java]: Resolved

File: src/main/java/org/opensrp/web/rest/SearchResource.java [hunk 3]
```diff
@@ -77,24 +75,33 @@ public List<Client> search(HttpServletRequest request) throws ParseException {//
 			searchBean.setLastEditFrom(lastEdit[0]);
 			searchBean.setLastEditTo(lastEdit[1]);
 		}
+
 		Map<String, String> attributeMap = null;
 		String attributes = getStringFilter("attribute", request);
```
[the remark this answers is not in the document]
Comment (SebaMutuku) [reply src/main/java/org/opensrp/web/rest/SearchResource.java]: Resolved

File: src/main/java/org/opensrp/web/rest/SearchResource.java [hunk 4]
```diff
@@ -77,24 +75,33 @@ public List<Client> search(HttpServletRequest request) throws ParseException {//
...
+		}
+		if (alternateName.isPresent()) {
+			attributeMap = new HashMap<>();
+			attributeMap.put(ALT_NAME, alternateName.get());
+		}
 		searchBean.setAttributes(attributeMap);
-		
+
 		Map<String, String> identifierMap = null;
 		String identifiers = getStringFilter("identifier", request);
```
[the remark this answers is not in the document]
Comment (SebaMutuku) [reply src/main/java/org/opensrp/web/rest/SearchResource.java]: Resolved

File: src/main/java/org/opensrp/web/rest/SearchResource.java (hunk 1 above)
Comment (SebaMutuku) [reply src/main/java/org/opensrp/web/rest/SearchResource.java]: Resolved

**Review by dubdabasoduba — APPROVED** [review]

**Review by dubdabasoduba — APPROVED** [review]

**Pull request merged**